File: clients/ios/SsdidDrive/SsdidDrive/Domain/Model/AuthProvider.swift
import Foundation

/// Authentication provider (OIDC IdP or WebAuthn)
struct AuthProvider: Codable, Identifiable {
    let id: String
    let name: String
    let providerType: String
    let enabled: Bool

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case providerType = "provider_type"
        case enabled
    }
}

/// User credential for WebAuthn or OIDC
struct UserCredential: Codable, Identifiable {
    let id: String
    let credentialType: String
    let name: String?
    let providerName: String?
    let createdAt: String
    let lastUsedAt: String?

    enum CodingKeys: String, CodingKey {
        case id
        case credentialType = "credential_type"
        case name
        case providerName = "provider_name"
        case createdAt = "created_at"
        case lastUsedAt = "last_used_at"
    }
}

File: clients/android/app/src/main/kotlin/my/ssdid/drive/domain/model/AuthProvider.kt
package my.ssdid.drive.domain.model

data class AuthProvider(
    val id: String,
    val name: String,
    val providerType: String,
    val tenantId: String,
    val clientId: String?,
    val issuer: String?,
    val enabled: Boolean
)

data class UserCredential(
    val id: String,
    val credentialType: String,
    val name: String?,
    val providerName: String?,
    val createdAt: String,
    val lastUsedAt: String?
)
